DEV_SETUP.md

```diff
@@ -30,8 +30,12 @@ Save those backups to somewhere you'll be able to access from the new environmen
 #### Prerequisites
 
 - [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
-- [Python 3.6](https://www.python.org/downloads/) and `python-dev`, `distutils` packages
-  - `sudo apt install python3-distutils python3-dev`
+- [Python 3.8](https://www.python.org/downloads/) and `python-dev`, `distutils` packages
+
+      $ sudo add-apt-repository ppa:deadsnakes/ppa
```

Comment:
Will we need to do this for our production servers? Quoting from https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa

> Disclaimer: If you want to use the package in a security-or-otherwise-critical environment (say, on a production server) , you do so at your own risk!

Do we need to be concerned about that? Looks like [we already use deadsnakes for 14.04](https://github.com/dimagi/commcare-cloud/blob/912a236b57e1a957cce30a0b3b73c058bfe8e695/src/commcare_cloud/ansible/roles/common_installs/tasks/main.yml#L15), so maybe not a big deal?

Reply:
> Looks like we already use deadsnakes for 14.04, so maybe not a big deal?

Interesting. I assumed commcare-cloud installed from Python.org. (Maybe it should?) Using Python Docker images for webworkers could be another option. (But maybe higher LOE to start doing, so less attractive.) Yeah, if we already use deadsnakes, maybe it's OK. But happy to have my mind changed here.

Reply:
One caution with using 3rd party PPAs for python is that we did get into a bit of a pickle in the past when the maintainer froze repo access to get funding:
https://forum.dimagi.com/t/issues-with-python-3-6-ppa-on-ubuntu-18-04-lts/7100

Reply:
> Yeah, if we already use deadsnakes, maybe it's OK.

Maybe... but in practice we are either very close to being or entirely off of 14.04, so probably not actually using it. I was thinking of the issue that Ethan highlighted when I wrote this (although couldn't remember the link). It would be good to use a supported installation channel if possible in light of that.

@ctsims @snopoke do either of you have any thoughts to add?

Reply:
I think if we haven't already, we should communicate that we are deprecating legacy support for Ubuntu 14, and pick a date in the future (probably out a ways, we need to give people months) when we can officially remove anything we are keeping around purely for Ubuntu 14 compatibility.

If this is a situation where Ubuntu 14 production systems will need to do something custom to continue running, but it's a one time package installation, I'm totally fine with doing that v. making a dependency for everyone running a modern OS.

Reply:
Sorry for not giving more context @ctsims. This is about using an Ubuntu repository (`deadsnakes/ppa`) that is unsupported for production. (It would make all of our VMs that have Python installed depend on that PPA rather than only Ubuntu 14 VMs.)

Reply:
(@ctsims does make a valid point though:

> I think if we haven't already, we should communicate that we are deprecating legacy support for Ubuntu 14

14.04 reached end of life in April last year.)

Reply:
fwiw, Ubuntu 20.04 LTS is scheduled to be released on April 23, and comes with Python 3.8.

Reply:
My view here is that we should avoid adding external dependencies if we can but if we have to we should make sure we're using a supported channel. Looking at the deadsnakes PPA they do have this prominent disclaimer:

> Disclaimer: there's no guarantee of timely updates in case of security problems or other issues. If you want to use them in a security-or-otherwise-critical environment (say, on a production server), you do so at your own risk.

Given that disclaimer I think I would be against using this in a production environment but fine in the context of this PR.

For rolling out 3.8 to live environments we should make sure that it is supported via the native OS channels or a channel that is committed to timely security updates etc.

Reply:
It looks like the [recommended approach](https://github.com/python/cpython/blob/3.8/README.rst#installing-multiple-versions) is to build from source, and to install with `make altinstall`, which will install the new binary alongside the version of Python that comes with your distro, without replacing it.

I agree that it seems OK to use the deadsnakes PPA for dev environments.

Building from source looks straight-forward. I think next steps would be to drop the deadsnakes PPA for Ubuntu 14.04 in commcare-cloud, and test building from source for both 14.04 and 18.04.